examples/src/lib.rs

```diff
@@ -199,21 +200,52 @@ fn extension_names() -> Vec<*const i8> {
     vec![
         Surface::name().as_ptr(),
         Win32Surface::name().as_ptr(),
-        DebugReport::name().as_ptr(),
+        DebugUtils::name().as_ptr(),
     ]
 }
 
 unsafe extern "system" fn vulkan_debug_callback(
-    _: vk::DebugReportFlagsEXT,
-    _: vk::DebugReportObjectTypeEXT,
-    _: u64,
-    _: usize,
-    _: i32,
-    _: *const c_char,
-    p_message: *const c_char,
-    _: *mut c_void,
-) -> u32 {
-    println!("{:?}", CStr::from_ptr(p_message));
+    message_severity: vk::DebugUtilsMessageSeverityFlagsEXT,
+    message_type: vk::DebugUtilsMessageTypeFlagsEXT,
+    p_callback_data: *const vk::DebugUtilsMessengerCallbackDataEXT,
+    _user_data: *mut std::os::raw::c_void,
+) -> vk::Bool32 {
+    let callback_data = *p_callback_data;
+
+    let message_severity = match message_severity {
+        vk::DebugUtilsMessageSeverityFlagsEXT::ERROR => String::from("ERROR"),
+        vk::DebugUtilsMessageSeverityFlagsEXT::WARNING => String::from("WARNING"),
+        vk::DebugUtilsMessageSeverityFlagsEXT::INFO => String::from("INFO"),
+        vk::DebugUtilsMessageSeverityFlagsEXT::VERBOSE => String::from("VERBOSE"),
+        _ => String::from("WARNING"),
+    };
```

Comment:
There's no need to allocate a `String` for these. Moreover, this roughly reimplements the existing `Debug` impl; better for an example to demonstrate use of that for improved concision, handling of unknown values, and free updates in the future.

Reply:
Done!